uwrt_mars_rover_hw/drivers/roboteq_driver/src/CanopenInterface.cpp

```diff
@@ -254,37 +232,38 @@ DataType CanopenInterface::sendQuery(RuntimeQuery query, uint8_t subindex) {
-    // TODO: throw error
-    return 0;
+    throw std::runtime_error("Written packet size does not match can frame size");
   }
 
   struct can_frame response_frame = {};
   ssize_t bytes_read = read(roboteq::CanopenInterface::socket_handle_, &response_frame, sizeof(struct can_frame));
 
-  //  std::cout << std::hex << response_frame.can_id << "\t" << static_cast<unsigned>(response_frame.can_dlc) << "\t"
-  //            << static_cast<unsigned>(response_frame.data[0]) << "\t" <<
-  //            static_cast<unsigned>(response_frame.data[1])
-  //            << "\t" << static_cast<unsigned>(response_frame.data[2]) << "\t"
-  //            << static_cast<unsigned>(response_frame.data[3]) << "\t" <<
-  //            static_cast<unsigned>(response_frame.data[4])
-  //            << "\t" << static_cast<unsigned>(response_frame.data[5]) << "\t"
-  //            << static_cast<unsigned>(response_frame.data[6]) << "\t" <<
-  //            static_cast<unsigned>(response_frame.data[7])
-  //            << std::endl;
+  ROS_DEBUG_STREAM(
+      std::hex << response_frame.can_id << "\t" << static_cast<unsigned>(response_frame.can_dlc) << "\t"
+               << static_cast<unsigned>(response_frame.data[0]) << "\t" << static_cast<unsigned>(response_frame.data[1])
+               << "\t" << static_cast<unsigned>(response_frame.data[2]) << "\t"
+               << static_cast<unsigned>(response_frame.data[3]) << "\t" << static_cast<unsigned>(response_frame.data[4])
+               << "\t" << static_cast<unsigned>(response_frame.data[5]) << "\t"
+               << static_cast<unsigned>(response_frame.data[6]) << "\t"
+               << static_cast<unsigned>(response_frame.data[7]));
 
   if (bytes_read != sizeof(struct can_frame)) {
-    // TODO: throw error
-    return 0;
+    throw std::runtime_error("Read packet size does not match can frame size");
+  }
+  if (static_cast<unsigned>(response_frame.can_dlc) != CAN_FRAME_SIZE_BYTES) {
+    throw std::runtime_error("Mismatched DLC value");
   }
-
-  // TODO: throw appropriate errors
   if ((response_frame.data[0] & RESPONSE_TYPE_MASK) != SUCCESSFUL_QUERY_RESPONSE) {
-    std::cout << "Query unsuccessful response" << std::endl;
-  } else if (query_frame.data[1] != response_frame.data[1] || query_frame.data[2] != response_frame.data[2]) {
-    std::cout << "Query response mismatched index" << std::endl;
-  } else if (query_frame.data[3] != response_frame.data[3]) {
-    std::cout << "Query response mismatched subindex" << std::endl;
-  } else {
+    throw std::runtime_error("Unssuccessful query response");
+  }
+  if (query_frame.data[1] != response_frame.data[1] || query_frame.data[2] != response_frame.data[2]) {
+    throw std::runtime_error("Mismatched index in query response");
+  }
+  if (query_frame.data[3] != response_frame.data[3]) {
+    throw std::runtime_error("Mismatched sub-index in query response");
+  }
+  // NOLINTNEXTLINE(readability-else-after-return)
```

Comment:
i dont think u should be ignoring lint here. I think just get rid of the else completely